File: frontend/src/dtos/brokerDto.ts
import type { BrokerAccessDto } from './broker/brokerAccessDto'
import type { BrokerSettingsDto } from './brokerSettingsDto'
import type { RabbitMQConnectionDto } from './RabbitMQConnectionDto'

export interface BrokerDto {
  id: string
  accessList: BrokerAccessDto[]
  system: string
  name: string
  rabbitMQConnection: RabbitMQConnectionDto
  settings: BrokerSettingsDto
  createdAt: string
  updatedAt: string
  syncedAt: string
  deletedAt: string
}

File: backend/ResQueue/ResQueue/Dtos/BrokerDto.cs
using ResQueue.Dtos.Broker;

namespace ResQueue.Dtos;

public record BrokerDto(
    string Id,
    string CreatedByUserId,
    List<BrokerAccessDto> AccessList,
    string System,
    string Name,
    RabbitMQConnectionDto? RabbitMQConnection,
    DateTime CreatedAt,
    DateTime UpdatedAt,
    DateTime? SyncedAt,
    DateTime? DeletedAt
);